crates/extensions/c8y_mapper_ext/src/converter.rs

```diff
@@ -273,6 +275,48 @@ impl CumulocityConverter {
         Ok(mqtt_messages)
     }
 
+    fn new_try_convert_measurement(
```

Comment:
What's the difference from the original `try_convert_measurement()`? Only how to get child ID?
Why not changing like this:

1. Create a new function `try_convert_measurement_with_maybe_id()` to take a `maybe_child_id` as an argument.
2. Call `try_convert_measurement_with_maybe_id()` inside `try_convert_measurement()`, so that we don't need to change anything on the function that call `try_convert_measurement()`.
3. Call `try_convert_measurement_with_maybe_id()` inside `new_try_convert_measurement()`.

Reply:
My plan was to keep converting the `telemetry` with new mqtt topics completely separate. Sooner or later we need to remove the older topics, then it will be much easier.

Reply:
Now the mapper supports only `te` topics. The `tedge to te` converter will handle messages on legacy topics and convert and republish them to corresponding `te` topics.